src/hooks/useSearchParamState.ts

```diff
@@ -0,0 +1,136 @@
+import { isDate } from 'date-fns';
+import { isNil } from 'lodash-es';
+import { useSearchParams } from 'react-router-dom';
+import { formatDateForGql, parseHmisDateString } from '@/modules/hmis/hmisUtil';
+
+// adapted from https://github.com/jschwindt/react-use-search-params-state/tree/main
```

Comment:
check in -- should we add this as an npm package to give proper credit

Reply:
I guess the pros are giving proper credit and being able to benefit from improvements to that repo. But the con is adding a dependency to something that seems to be maintained by just one person, and isn't very widely used. Could this be a security concern as well?

We could look at whether there are other npm packages that are more popular and by extension maybe more trustworthy, maybe [this](https://www.npmjs.com/package/use-query-params?activeTab=readme) provides similar functionality?

Reply:
leaving this as is. per theron's other comment, too, if we use it going forward we need to make it type safe. [ticket for that](https://www.pivotaltracker.com/story/show/187258786)